src/chains/vaults/chain.vaults.ts

```diff
@@ -45,12 +46,14 @@ export class ChainVaults {
       );
 
       try {
-        for await (const compoundVault of query) {
-          const cachedVaultIx = this.cachedVaults.findIndex((v) => v.address === compoundVault.address);
-          if (cachedVaultIx >= 0) {
-            this.cachedVaults[cachedVaultIx] = compoundVault;
+        for await (const vault of query) {
+          const index = this.cachedVaults.findIndex((v) => v.address === vault.address);
+          if (index >= 0) {
+            this.cachedVaults[index] = vault;
           } else {
-            this.cachedVaults.push(compoundVault);
+            if (vault.stage === Stage.Production || vault.stage === STAGE) {
```

Comment:
so this makes staging API have all vaults (prod and stag) and production just have Production ones?

Reply:
coorrect